### PR TITLE
feat(bot,types): Add MessageCall

### LINKED ISSUE #3595: [api-docs] add message.call field (#6864)
A new commit was made into the api-docs repo: https://github.com/discord/discord-api-docs/commit/f7393bd4f8f34103ad36d2ba6ccd5ec0eaa97bf7

This is a bot created issue.

### PR DESCRIPTION
**Pull request by Fleny113**
Add MessageCall, this object will be present on the call message in (Group) DMs 

Upstream: [discord/discord-api-docs#6864](https://github.com/discord/discord-api-docs/pull/6864)
fixes #3595

### CODE REVIEW

File: packages/bot/src/transformers/message.ts [hunk 1]
```diff
@@ -415,6 +419,16 @@ export function transformMessageInteractionMetadata(bot: Bot, payload: DiscordMe
   return bot.transformers.customizers.messageInteractionMetadata(bot, payload, metadata)
 }
 
+export function transformMessageCall(bot: Bot, payload: DiscordMessageCall): MessageCall {
+  const call = {} as MessageCall
+  const props = bot.transformers.desiredProperties.message.call
+
+  if (props.participants && payload.participants.length > 0) call.participants = payload.participants.map((x) => bot.transformers.snowflake(x))
```
Comment (lts20050703) [inline packages/bot/src/transformers/message.ts:426]:
```suggestion
  if (props.participants && payload.participants) call.participants = payload.participants.map((x) => bot.transformers.snowflake(x))
```
Why not keep the empty array as-is?

Comment (Fleny113) [reply packages/bot/src/transformers/message.ts:426]: depends on if we want to give the user an empty array or undefined, for me it is fine just removing the length check, but we do have it in other places

Comment (lts20050703) [reply packages/bot/src/transformers/message.ts:426]: Personally I think that if the user desire the participants property, they should get an empty array

**Review by lts20050703 — APPROVED** [review]
LGTM!

File: packages/bot/src/transformers.ts [hunk 2]
```diff
@@ -380,6 +389,10 @@ export interface Transformers {
       thread: boolean
       type: boolean
       webhookId: boolean
+      call: {
+        participants: boolean
+        endedTimestamp: boolean
```
Comment (AwesomeStickz) [inline packages/bot/src/transformers.ts:394]:
```suggestion
        endedTimestamp: number
```

**Review by AwesomeStickz — CHANGES_REQUESTED** [review]

File: packages/bot/src/transformers.ts [hunk 3]
```diff
@@ -979,6 +996,10 @@ export function createTransformers(options: Partial<Transformers>, opts?: Create
         thread: opts?.defaultDesiredPropertiesValue ?? false,
         type: opts?.defaultDesiredPropertiesValue ?? false,
         webhookId: opts?.defaultDesiredPropertiesValue ?? false,
+        call: {
+          participants: opts?.defaultDesiredPropertiesValue ?? false,
+          endedTimestamp: opts?.defaultDesiredPropertiesValue ?? false,
```
[the remark this answers is not in the document]
Comment (lts20050703) [reply packages/bot/src/transformers.ts:1001]: @AwesomeStickz Sticks... that's desiredProperties...

File: packages/bot/src/transformers.ts (hunk 2 above)
Comment (lts20050703) [reply packages/bot/src/transformers.ts:394]: @AwesomeStickz This is also desiredProperties...

File: packages/bot/src/transformers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (AwesomeStickz) [reply packages/bot/src/transformers.ts:1001]: ty for pointing it out

**Review by AwesomeStickz — APPROVED** [review]

**Pull request merged**